pkg/netlink/netlink_watcher.go

```diff
@@ -0,0 +1,1988 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
+
+// Package netlink handles the netlink related functionality
+package netlink
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"encoding/binary"
+	"encoding/json"
+	"net"
+	"reflect"
+
+	"golang.org/x/sys/unix"
+
+	vn "github.com/vishvananda/netlink"
+
+	"path"
+
+	"github.com/opiproject/opi-evpn-bridge/pkg/config"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb"
+	eb "github.com/opiproject/opi-evpn-bridge/pkg/netlink/eventbus"
+	"github.com/opiproject/opi-evpn-bridge/pkg/utils"
+)
+
+var ctx context.Context
+var nlink utils.Netlink
+
+// dbLock variable
+// var dbLock int
+
+// GRD variable
+var GRD int
+
+// pollInterval variable
+var pollInterval int
+
+// dbphyPortslock variable
+var phyPorts = make(map[string]int)
+
+// stopMonitoring variable
+var stopMonitoring bool
+
+// EventBus variable
+var EventBus = eb.NewEventBus()
+
+// strNone variable
+var strNone = "NONE"
+var zebraStr = "zebra"
+
+// Route Direction
+const ( // Route direction
+	None int = iota
+	RX
+	TX
+	RXTX
+)
+
+// Nexthop type
+const ( // NexthopStruct TYPE & L2NEXTHOP TYPE & FDBentry
+	PHY = iota
+	SVI
+	ACC
+	VXLAN
+	BRIDGEPORT
+	OTHER
+)
+
+// RTNNeighbor
+const (
+	RTNNeighbor = 1111
+)
+
+// NeighKey strcture of neighbor
+type NeighKey struct {
+	Dst     string
+	VrfName string
+	Dev     int
+}
+
+// RouteKey structure of route description
+type RouteKey struct {
+	Table int
+	Dst   string
+}
+
+// NexthopKey structure of nexthop
+type NexthopKey struct {
+	VrfName string
+	Dst     string
+	Dev     int
+	Local   bool
+}
+
+// NeighIPStruct nighbor ip structure
+type NeighIPStruct struct {
+	Dst         string
+	Dev         string
+	Lladdr      string
+	ExternLearn string
+	State       []string
+	Protocol    string
+}
+
+// FDBKey structure key for sorting theFDB entries
+type FDBKey struct {
+	VlanID int
+	Mac    string
+}
+
+// L2NexthopKey is l2 neighbor key
+type L2NexthopKey struct {
+	Dev    string
+	VlanID int
+	Dst    string
+}
+
+// FdbIPStruct fdb ip structure
+type FdbIPStruct struct {
+	Mac    string
+	Ifname string
+	Vlan   int
+	Flags  []string
+	Master string
+	State  string
+	Dst    string
+}
+
+// Routes Variable
+var Routes = make(map[RouteKey]RouteStruct)
+
+// Nexthops Variable
+var Nexthops = make(map[NexthopKey]NexthopStruct)
```

Comment:
we dont have any pointers in key.

Reply:
it is a warning. Can you please
1 double check
2 find maybe a linter which would run compile time error in case of a ptr/slice/map member was added to struct used as a key

Reply:
verified. Golangci-lint does have the check for this.

Reply:
do we have it enabled for this repo? Pls post the link

Reply:
https://github.com/opiproject/opi-evpn-bridge/blob/main/.golangci.yml#L21

Reply:
which exactly check is responsible for that?

Reply:
staticcheck , govet and gosec .